From 8190b0c3834af4e3c41a93fd6853b3bfbdc2295f Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Mon, 11 Sep 2023 15:32:38 +0200
Subject: [PATCH 02/13] configurations: sbp1: Add M.2 NVMe temperature sensors

SBP1 has two M.2 slots that hosts NVMe drives. When a NVMe is pluged
the temperature can be read using nvmesensor.

Add a custom name for each M.2 drive and add it to a separate
temperature zone.

Tested: M.2 NVMe temperatures are not visible on the webui as the M.2
        drives do not have a provision IPMI FRU EEPROM.

        busctl introspect xyz.openbmc_project.NVMeSensor \
        /xyz/openbmc_project/sensors/temperature/M2_SSD2_Temperature
        shows the NVMe temperature.

Change-Id: I3449450069329fb9241ca5a3cff4feba62e13555
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/sbp1_chassis.json | 26 ++++++++++++++++++++++++++
 configurations/sbp1_nvme.json    | 20 ++++++++++++++++++++
 meson.build                      |  1 +
 3 files changed, 47 insertions(+)
 create mode 100644 configurations/sbp1_nvme.json

diff --git a/configurations/sbp1_chassis.json b/configurations/sbp1_chassis.json
index 92490aa..a9c9c9b 100644
--- a/configurations/sbp1_chassis.json
+++ b/configurations/sbp1_chassis.json
@@ -2021,6 +2021,32 @@
                 "PSU"
             ]
         },
+        {
+            "Class": "temp",
+            "FFGainCoefficient": 0,
+            "FFOffCoefficient": 0,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
+            "ILimitMin": 2500,
+            "Inputs": [
+                "M2 SSD1 Temperature",
+                "M2 SSD2 Temperature"
+            ],
+            "Name": "M2 SSD Temperature",
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
+            "OutLimitMin": 2500,
+            "Outputs": [],
+            "PCoefficient": -500,
+            "PositiveHysteresis": 0,
+            "SetPoint": 85,
+            "SlewNeg": 0,
+            "SlewPos": 0,
+            "Type": "Pid",
+            "Zones": [
+                "PSU"
+            ]
+        },
         {
             "FailSafePercent": 35,
             "MinThermalOutput": 2500,
diff --git a/configurations/sbp1_nvme.json b/configurations/sbp1_nvme.json
new file mode 100644
index 0000000..a5a329d
--- /dev/null
+++ b/configurations/sbp1_nvme.json
@@ -0,0 +1,20 @@
+{
+    "Exposes": [
+        {
+            "Address": "0x6a",
+            "Bus": "50",
+            "Name": "M2 SSD1 Temperature",
+            "Type": "NVME1000"
+        },
+        {
+            "Address": "0x6a",
+            "Bus": "52",
+            "Name": "M2 SSD2 Temperature",
+            "Type": "NVME1000"
+        }
+    ],
+    "Name": "SBP1 NVMe",
+    "Probe": "xyz.openbmc_project.FruDevice({'BOARD_PRODUCT_NAME': '.*SBP1'})",
+    "ProductId": 0,
+    "Type": "NVMe"
+}
diff --git a/meson.build b/meson.build
index 5cd6abb..42d00c0 100644
--- a/meson.build
+++ b/meson.build
@@ -144,6 +144,7 @@ configs = [
     'sas_module.json',
     'sbp1_baseboard.json',
     'sbp1_chassis.json',
+    'sbp1_nvme.json',
     'sbp1_psu.json',
     'system1_baseboard.json',
     'solum_pssf162202_psu.json',
-- 
2.41.0

